core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -1045,10 +1080,12 @@ private void advanceState() {
             int nextIndex = chunkIter.next();
             state.set(chunk, chunkIter, nextIndex);
 
-            ByteBuffer key = state.getChunk().readKey(state.getIndex());
-            if (!inBounds(key)) {
-                state = null;
-                return;
+            if (((hi != null) && !isDescending) || ((lo != null) && isDescending)) {
```

Comment:
Please comment that this is an optimization. 

IMHO, would be good to remove the parenthesis around (hi != null) and (lo != null)